src/main/kotlin/net/maizegenetics/phgv2/utils/SeqUtils.kt

```diff
@@ -9,36 +9,195 @@ import java.util.logging.Logger
+ *   Returns: Map<String, NucSeq> where the key is the genome and the value is a Biokotlin NucSeq
+ *
+ * 3. retrieveAgcData: called to pull data using    the "agc listset" or "agc listctg" command.
+ *   Expected user input:  dbPath:String, agcCmdList:List<String>
+ *   a.  For listset: the user agcCmdList should contain a single item: "listset"
+ *       This command  returns a list of all the genomes in the AGC file.
+ *   b.  For listctg: the user agcCmdList should contain "listctg" as its first entry.
+ *       The remaining entries should be genome names whose contigs will be listed.
+ *       If no genomes names are supplied, an error is thrown
+ *   Result:  a list of strings, one per genome, in the format:
+ *      genomeName:ctg1,ctg2,ctg3,ctg4
+ *      e.g. LineA:1,2
+ *           LineB:1,2
+ *
+ * The rest of the functions in this file are used internally by the above functions.
+ *
  */
 fun retrieveAgcContigs(dbPath:String,ranges:List<String>):Map<String, NucSeq> {
 
-    val commands = buildAgcCommandFromList(dbPath, ranges)
+    val commands = buildAgcCommandFromList(dbPath, "getctg",ranges)
+    val chromNucSeqMap = queryAgc(commands)
+    return chromNucSeqMap
+}
+
+fun retrieveAgcGenomes(dbPath:String,genomes:List<String>):Map<String, NucSeq> {
+
+    val commands = buildAgcCommandFromList(dbPath, "getset",genomes)
     val chromNucSeqMap = queryAgc(commands)
     return chromNucSeqMap
 }
 
+/**
+ * Processes AGC commands listset or listctg
+ */
+fun retrieveAgcData(dbPath:String,agcCmdList:List<String>):List<String>? {
+
+    val agcFile = "${dbPath}/assemblies.agc"
+
+    check(File(agcFile).exists()) { "File assemblies.agc does not exist in folder $dbPath- please send a valid path that indicates the parent folder for your assemblies.agc compressed file." }
+    check(agcCmdList.isNotEmpty()) { "No AGC command was sent to retrieveAgcData" }
+
+    val agcCmd = agcCmdList[0]
+    val genomes = mutableListOf<String>()
+
+    if (agcCmd == "listctg") {
+        check(agcCmdList.size > 1) { "No genomes were sent to retrieveAgcData " }
+        genomes.addAll(agcCmdList.subList(1,agcCmdList.size))
+    }
+
+    val command = mutableListOf<String>()
+    val commandPrefix = arrayOf("conda","run","-n","phgv2-conda","agc",agcCmd,agcFile)
+    command.addAll(commandPrefix)
+    command.addAll(genomes)
+
+    val agcProcess = ProcessBuilder(*command.toTypedArray())
+        .start()
+
+    val agcOut = BufferedInputStream(agcProcess.inputStream, 5000000)
+    try {
+        // check command.  Should only be listctg or listset.
+        // call appropriate function to process the output
+        val queryData = when (agcCmd) {
+            "listctg" -> {
+                dataFromListCtgs(agcOut, genomes)
+            }
+            "listset" -> {
+                dataFromListSet(agcOut)
+            }
+            else -> {
+                myLogger.severe("Invalid AGC command sent to retrieveAgcData: $agcCmd")
```

Comment:
Couldn't this just be a myLogger.error?

Reply:
Yes, well that is the whole logger issue.  "error" doesn't show up as an option, perhaps because I'm using: (copied from somewhere initially)

private val myLogger = Logger.getLogger("net.maizegenetics.phgv2.utils.SeqUtils")

THis is from import java.util.logging.Logger

I thought that "private val myLogger = LogManager.getLogger(AgcCompress::class.java)" required a class, and the SeqUtils.kt is not a class.

I'm happy to change this however you think it can be changed.  I know the code in this repository has inconsistent logging methods.

Reply:
If its not letting you do an error this is probably fine.  Will still throw an exception and stop processing.

Reply:
Yes, it compiles.  It is used Logger.getLogger() is used in CreateRefVcf.kt, VariantLoadingUtils.kt and SeqUtils.kt.  The other classes in clikt are using LogManager.getLogger().  I can see about changing this

Reply:
That must be coming from another dependency. Our build.gradle.kts has a dependency on Log4j2

Reply:
I changed it to use apache LogManager as with the cli classes.